src/Core/Grid/Query/CatalogPriceRuleQueryBuilder.php

```diff
@@ -0,0 +1,210 @@
+            pr_country.name country,
+            pr_group.name group_name'
+        );
+        $this->searchCriteriaApplicator
+            ->applyPagination($searchCriteria, $qb)
+            ->applySorting($searchCriteria, $qb)
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters())
+            ->select('COUNT(DISTINCT pr.`id_specific_price_rule`)')
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * Gets query builder with the common sql for catalog price rule listing.
+     *
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder(array $filters)
+    {
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'specific_price_rule', 'pr')
+            ->leftJoin(
+                'pr',
+                $this->dbPrefix . 'shop',
+                'pr_shop',
+                'pr_shop.`id_shop` = pr.`id_shop` AND pr.`id_shop` IN (:contextShopIds)'
+            )
+            ->leftJoin(
+                'pr',
+                $this->dbPrefix . 'currency_lang',
+                'pr_currency',
+                'pr_currency.`id_currency` = pr.`id_currency` AND pr_currency.`id_lang` = :contextLangId'
+            )
+            ->leftJoin(
+                'pr',
+                $this->dbPrefix . 'country_lang',
+                'pr_country',
+                'pr_country.`id_country` = pr.`id_country` AND pr_country.`id_lang` = :contextLangId'
+            )
+            ->leftJoin(
+                'pr',
+                $this->dbPrefix . 'group_lang',
+                'pr_group',
+                'pr_group.`id_group` = pr.`id_group` AND pr_group.`id_lang` = :contextLangId'
+            )
+        ;
+
+        $this->applyFilters($qb, $filters);
+        $qb->setParameter('contextLangId', $this->contextIdLang);
+        $qb->setParameter('contextShopIds', $this->contextShopIds, Connection::PARAM_INT_ARRAY);
+
+        return $qb;
+    }
+
+    /**
+     * @param QueryBuilder $qb
+     * @param array $filters
+     */
+    private function applyFilters(QueryBuilder $qb, array $filters)
+    {
+        $allowedFiltersAliasMap = [
+            'id_specific_price_rule' => 'pr.id_specific_price_rule',
+            'name' => 'pr.name',
+            'from_quantity' => 'pr.from_quantity',
+            'reduction' => 'pr.reduction',
+            'reduction_type' => 'pr.reduction_type',
+            'from' => 'pr.from',
+            'to' => 'pr.to',
+            'shop' => 'pr_shop.name',
+            'currency' => 'pr_currency.name',
+            'country' => 'pr_country.name',
+            'group_name' => 'pr_group.name',
+        ];
+
+        $exactMatchFilters = ['id_specific_price_rule', 'from_quantity', 'reduction', 'reduction_type'];
```

Comment:
@matks, according to legacy- reduction field in list is shown rounded to decimals, so filtering float which is, for example 50.0005 wont find it as exact match, but user will see 50.00 in unfiltered list. Seems like UX issue to me. 
Currently i just didn't round the numbers. Any other offers?

Reply:
@eternoendless This is a Grid migration PR. Grid do not depend on legacy code, only on the legacy SQL schema.

I think we can introduce https://github.com/PrestaShop/decimal here. Do you agree ?

Reply:
Use Decimal as much as you can, specially when performing price operations.

Reply:
Also, when rounding, you must keep in mind the rounding mode as configured in the shop settings.

Reply:
according to https://github.com/PrestaShop/PrestaShop/issues/12223 I guess CLDR should handle the rounding.

Reply:
Sorry @zuk3975 my answer is irrelevant
I think this is an issue that must be discussed further with UX and Product, in another PR. I added it to the todo-list https://github.com/PrestaShop/PrestaShop/issues/10548 so we wont forget

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/18691 😄  almost one year later